src/openni2_driver.cpp

```diff
@@ -59,6 +59,8 @@ OpenNI2Driver::OpenNI2Driver(ros::NodeHandle& n, ros::NodeHandle& pnh) :
 
   readConfigFromParameterServer();
 
+  device_manager_->registerReconnectCb(&OpenNI2Driver::reconnectDevice);
```

Comment:
Changing this line to `
  device_manager_->registerReconnectCb( std::tr1::bind(&OpenNI2Driver::reconnectDevice, this) );` fixes compile error (no clue if this is the answer, or if it works).